cli/pkg/kctrl/cmd/package/builder/package-build.go

```diff
@@ -0,0 +1,38 @@
+package builder
+
+import (
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type PackageBuild struct {
+	metav1.TypeMeta `json:",inline"`
+	Spec            Spec `json:"spec, omitempty"`
+}
+
+type Imgpkg struct {
+	metav1.TypeMeta  `json:",inline"`
+	RegistryURL      string `json:"registryUrl"`
+	RegistryUserName string `json:"registryUserName"`
+	RegistryPassword string `json:"registryPassword"`
```

Comment:
cannot store username/password (or any other auth material) in a "non secured area". credentials would have to be provided by the user every time (most likely via env vars, or already something like docker login)

Reply:
Removed them from the struct. Also added a line to do docker login while giving registry details.
`createImgPkgStep.ui.BeginLinef("To push the image onto registry, ensure that `docker login` is done onto `%s`. If not done, open a separate tab and run `docker login %s` and enter the valid credentials to login successfully.", registryAuthDetails.RegistryURL, registryAuthDetails.RegistryURL)`